anndata/core/anndata.py

```diff
@@ -1011,6 +1011,12 @@ def layers(self, value):
             self._init_as_actual(self.copy())
         self._layers = layers
 
+    @layers.deleter
+    def layers(self):
+        if self.isview:
+            self._init_as_actual(self.copy())
```

Comment:
I see those lines repeated a lot. How about a decorator `actualizes_view` or so?

```py
@layers.deleter
@actualizes_view
def layers(self):
    self._layers = Layers(self, vals=dict())
```

Reply:
I was thinking about that while I wrote this. I would prefer a private method, not a decorator, so you can validate the input before calling `_init_as_actual`. Many of the `setter` have examples of this.

The only other thing is that it might be better to modify the copy, then call `_init_as_actual`, with it, just to make sure all possible checks occurred before modifying the object. This is harder to put in a method.

Reply:
Thoughts on this?

If it were to be a method, how about this being the no-argument version of `self._init_as_actual()`? The only thing I'm worried about there is whether's it's obvious that it would be a no-op if the object isn't a view.

Reply:
That would make sense! Just make `_init_as_actual` work without arguments and make it do nothing if `self` is actualized.

Reply:
I do feel bad every time I make the `_init_as_*` methods more complicated. Someday they'll be refactored. Someday...

Reply:
Ehh. I'm going back on this. Reading this code:

```python
    @varm.deleter
    def varm(self):
        self._init_as_actual()
        self._varm = AxisArrays(self, 1, vals=dict())
```

I wouldn't expect that `self._init_as_actual()` was dependent on whether the object was a view. I think adding more logic to `_init_as_actual` hurts maintainability. I'm pushing simplified `deleter` methods though.